### PR TITLE
References public feedback

### PR DESCRIPTION
**Pull request by nicksellen**
#### Proposed Changes

* add `feedbackPublic` field for references
* make `recommend` field private

#### Testing Instructions

Fixes #

### CODE REVIEW

File: modules/references/tests/server/reference-create.server.routes.tests.js [hunk 1]
```diff
@@ -72,6 +74,7 @@ describe('Create a reference', () => {
                 hostedThem: true,
               },
               recommend: 'yes',
```
Comment (edvardmk) [inline modules/references/tests/server/reference-create.server.routes.tests.js:76]: should this still be included?

Comment (nicksellen) [reply modules/references/tests/server/reference-create.server.routes.tests.js:76]: Yes, as the user will still submit that info, it's just that we won't return it.

File: modules/references/tests/server/reference-create.server.routes.tests.js [hunk 2]
```diff
@@ -605,6 +608,28 @@ describe('Create a reference', () => {
         });
       });
 
+      it.only('[too long public feedback] 400', async () => {
```
Comment (mrkvon) [inline modules/references/tests/server/reference-create.server.routes.tests.js:611]: remove `.only`

File: modules/references/server/controllers/reference.server.controller.js [hunk 3]
```diff
@@ -72,8 +73,23 @@ function validateCreate(req) {
     details.userTo = 'userId expected';
   }
 
+  if (_.has(req, ['body', 'feedbackPublic'])) {
+    const { feedbackPublic } = req.body;
```
Comment (nicksellen) [inline modules/references/server/controllers/reference.server.controller.js:77]: Comment from @simison in https://github.com/Trustroots/trustroots/pull/1384#discussion_r409512920

> What kind of sanitation we should add here?

I borrowed the client side part from the report message, which is plaintext and uses `textService.plainText`, so seems a good option here too.

**Comment by nicksellen** [comment]
Thanks for review!

**Pull request merged**